CRM/Activity/Form/ActivityLinks.php

```diff
@@ -78,8 +78,22 @@ public static function commonBuildQuickForm($self) {
         }
         // Check for existence of a mobile phone and ! do not SMS privacy setting
         $mobileTypeID = CRM_Core_PseudoConstant::getKey('CRM_Core_BAO_Phone', 'phone_type_id', 'Mobile');
-        list($name, $phone, $doNotSMS) = CRM_Contact_BAO_Contact_Location::getPhoneDetails($contactId, $mobileTypeID);
-        if (!$doNotSMS && $phone) {
+        try {
+          $phone = civicrm_api3('Phone', 'getsingle', array(
+            'contact_id' => $contactId,
+            'phone_type_id' => $mobileTypeID,
+            'return' => array('phone', 'contact_id'),
+            'options' => array('limit' => 1, 'sort' => "is_primary DESC"),
+            'api.Contact.getsingle' => array(
+              'id' => '$value.contact_id',
+              'return' => 'do_not_sms',
+            ),
+          ));
+        }
+        catch (CiviCRM_API3_Exception $e) {
+          continue;
+        }
+        if (!$phone['api.Contact.getsingle']['do_not_sms'] && $phone['phone']) {
```

Comment:
I hope this doesn't throw notice errors as it will hit ```continue``` at line 94 if $phone is empty

Reply:
If the contact does not have a phone, then getsingle will indeed throw an exception which will get caught, so no notice will be shown. I ran the code to check that this actually happened.

Reply:
Cool 👍